src/BuiltInTools/dotnet-watch/HotReload/BlazorWebAssemblyDeltaApplier.cs

```diff
@@ -123,33 +113,18 @@ public async ValueTask ReportDiagnosticsAsync(DotNetWatchContext context, IEnume
         private async Task<bool> VerifyDeltaApplied(DotNetWatchContext context, CancellationToken cancellationToken)
         {
             var _receiveBuffer = new byte[1];
-            try
+            var result = await context.BrowserRefreshServer!.ReceiveAsync(_receiveBuffer, cancellationToken);
+            if (result is null)
             {
-                // We want to give the client some time to ACK the deltas being applied. VerifyDeltaApplied is limited by a
-                // 5 second wait timeout enforced using a WaitAsync. However, WaitAsync only works reliably if the calling
-                // function is async. If BrowserRefreshServer.ReceiveAsync finishes synchronously, the WaitAsync would
-                // never have an opportunity to execute. Consequently, we'll give it some reasonable number of opportunities
-                // to loop before we decide that applying deltas failed.
-                for (var i = 0; i < 100; i++)
-                {
-                    var result = await context.BrowserRefreshServer!.ReceiveAsync(_receiveBuffer, cancellationToken);
-                    if (result is null)
-                    {
-                        // A null result indicates no clients are connected. No deltas could have been applied in this state.
-                        _reporter.Verbose("No client is connected to ack deltas");
-                        return false;
-                    }
-
-                    if (IsDeltaReceivedMessage(result.Value))
-                    {
-                        // 1 indicates success.
-                        return _receiveBuffer[0] == 1;
-                    }
-                }
+                // A null result indicates no clients are connected. No deltas could have been applied in this state.
+                _reporter.Verbose("No client is connected to ack deltas");
```

Comment:
Should this be reported as verbose? Wouldn't it be better as a warning/error. If I understand this correctly, at this point things are in a broken state, aren't they?

Reply:
ReceiveAsync already reports any errors.

I'll clean this up a bit later on. It's not clear to me why this [catch](https://github.com/dotnet/sdk/blob/main/src/BuiltInTools/dotnet-watch/Filters/BrowserRefreshServer.cs#L225) is in the loop over `_clientSockets`. I would expect it to be outside of the loop. I'd expect the semantics of the method be: find the first socket that's open and use it to receive data. If it fails then don't try another one and just fail the whole thing.